### PR TITLE
Add DUPICATE_REMOVAL operator to bags

### PR DESCRIPTION
**Pull request by mudathirmahgoub**
This PR adds duplicate removal operator to bags (also known as delta or squash). 
Other changes:
- print MK_BAG operator as "bag" in smt2 instead of "mkBag"
- renamed BAG_IS_INCLUDED operator to SUBBAG.

### CODE REVIEW

File: src/api/cvc4cppkind.h [hunk 1]
```diff
@@ -2047,6 +2048,16 @@ enum CVC4_PUBLIC Kind : int32_t
...
+  /**
+   * Eliminate duplicates in a given bag. The returned bag contains exactly the
+   * same elements in the given bag, but with multiplicity one.
+   * Parameters: 1
+   *   -[1]: a term of bag sort
+   * Create with:
+   *   mkTerm(Kind kind, Term child)
+   *   mkTerm(Kind kind, const std::vector<Term>& children)
+   */
+  DUPLICATE_REMOVAL,
```
Comment (ajreynol) [inline src/api/cvc4cppkind.h:2060]: Should we call this BAG_DUPLICATE_REMOVAL?

**Review by ajreynol — CHANGES_REQUESTED** [review]
One minor request

File: src/api/cvc4cppkind.h (hunk 1 above)
Comment (mudathirmahgoub) [reply src/api/cvc4cppkind.h:2060]: `BAG_DUPLICATE_REMOVAL` is a bit long. I prefer `DUPLICATE_REMOVAL` as long as there is no confusion with other theories.

Comment (mudathirmahgoub) [reply src/api/cvc4cppkind.h:2060]: Another alternatives I found are `squash`, `Duplicate elimination`, and recently `Deduplication`

**Review by ajreynol — APPROVED** [review]

**Pull request merged**